datashader/transfer_functions/_cuda_utils.py

```diff
@@ -70,13 +72,35 @@ def masked_clip_2d(data, mask, lower, upper):
     masked_clip_2d_kernel[cuda_args(data.shape)](data, mask, lower, upper)
 
 
+# Behaviour of numba.cuda.atomic.max/min changed in 0.50 so as to behave as per
+# np.nanmax/np.nanmin
+if LooseVersion(numba.__version__) >= LooseVersion("0.51.0"):
+    @cuda.jit
+    def cuda_atomic_nanmin(ary, idx, val)
+        return cuda.atomic.nanmin(ary, idx, val)
+    @cuda.jit
+    def cuda_atomic_nanmax(ary, idx, val)
+        return cuda.atomic.nanmax(ary, idx, val)
+elif LooseVersion(numba.__version__) <= LooseVersion("0.49.0"):
+    @cuda.jit
+    def cuda_atomic_nanmin(ary, idx, val)
+        return cuda.atomic.min(ary, idx, val)
+
+    @cuda.jit
+    def cuda_atomic_nanmax(ary, idx, val)
+        return cuda.atomic.max(ary, idx, val)
```

Comment:
Are the `def`s really needed here?  Would it not be sufficient to do `cuda_atomic_nanmin = cuda.atomic.nanmin` in the first `if` and `cuda_atomic_nanmin = cuda.atomic.min` in the second?

Reply:
Unfortunately I think they are (and these may need to be device functions too), the `cuda.atomic.max/min` are stubs that are basically a token known to the compiler, if they appear as globals I don't think the compiler will know what to do with them.

Reply:
Ok, thanks.